### PR TITLE
feat(new tool): Regex Tester (and Cheatsheet)

### PR DESCRIPTION
**Pull request by sharevb**
Fix https://github.com/CorentinTh/it-tools/issues/1007, https://github.com/CorentinTh/it-tools/issues/991, https://github.com/CorentinTh/it-tools/issues/936, https://github.com/CorentinTh/it-tools/issues/761, https://github.com/CorentinTh/it-tools/issues/649 https://github.com/CorentinTh/it-tools/issues/644, https://github.com/CorentinTh/it-tools/issues/554 https://github.com/CorentinTh/it-tools/issues/308, #1293

### CODE REVIEW

**Comment by sharevb** [comment]
Hi @CorentinTh , could you consider merging, please ? thanks

**Comment by sharevb** [comment]
Hi @steffenrapp, thanks for this bug report, it should be ok now (ctrl f5) ; by the way, found what caused white square but didn't reproduce 'disabled' checkbox in chrome windows.

**Comment by steffenrapp** [comment]
Thank you @sharevb 😃 now both bugs are fixed for me. Hope it will be merged soon!

**Comment by steffenrapp** [comment]
Hi @sharevb on more thing from my side ... To me it feels kind of wrong to have an external link in the tool. I think no other tool has that currently and users can search for that if they need it. Let me know what you think.

**Comment by sharevb** [comment]
Hi @steffenrapp, I put a link to https://www.regular-expressions.info/javascript.html because it is the up to date documentation of JS regex syntaxe, but may be I can add a "JS Regex Memo" tool (as there are other memo, ie markdown) but I need to find a free tuto of regex syntaxes

**Comment by sharevb** [comment]
Hi @steffenrapp, added a Regex Cheatsheet with the regex tester (and linked it instead of external)

**Comment by steffenrapp** [comment]
@sharevb I like that! 👍🏼

**Comment by steffenrapp** [comment]
Hi @CorentinTh I don't know how you have the resources to review the PRs at the moment, but it seems that a lot of people are waiting for some of the features that are in the pipeline, especially this one. Maybe it makes sense to have someone else help you with merging the PRs who has the same quality goals as you do for this project, to speed up development? 😃 Would be great to see many more new helpful tools in the final product.

File: src/tools/regex-memo/index.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import { defineTool } from '../tool';
+
+export const tool = defineTool({
+  name: 'Regex cheatsheet',
+  path: '/regex-memo',
+  description: 'Javascript Regex/Regular Expression cheatsheet',
+  keywords: ['regex', 'regular', 'expression', 'javascript', 'memo', 'cheatsheet'],
+  component: () => import('./regex-memo.vue'),
+  icon: BrandJavascript,
+  createdAt: new Date('2024-09-14'),
```
Comment (CorentinTh) [inline src/tools/regex-memo/index.ts:11]:
```suggestion
  createdAt: new Date('2024-09-20'),
```

**Review by CorentinTh — DISMISSED** [review]

File: src/tools/regex-tester/index.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import { defineTool } from '../tool';
+
+export const tool = defineTool({
+  name: 'Regex Tester',
+  path: '/regex-tester',
+  description: 'Regex Tester',
+  keywords: ['regex', 'tester', 'sample', 'expression'],
+  component: () => import('./regex-tester.vue'),
+  icon: Language,
+  createdAt: new Date('2024-09-14'),
```
Comment (CorentinTh) [inline src/tools/regex-tester/index.ts:11]:
```suggestion
  createdAt: new Date('2024-09-20'),
```

**Comment by CorentinTh** [comment]
Hi @sharevb and @steffenrapp 
Thank you for your feedback and consideration. I understand that many are waiting for new features, and I appreciate your patience. Maintaining the project's quality is very important to me, and I'm working to review the PRs as efficiently as possible. I'm also considering adding a maintainer to help speed up development while upholding our quality standards.

**Comment by steffenrapp** [comment]
Hi @CorentinTh Thank you for reviewing which is a lot of work if you want to do it properly. I also think quality should always be top priority to create a stable product. Nobody likes to use buggy tools that you can't rely on when you use them on a regular basis.
@sharevb Thanks for all the work you put into contributing to IT Tools!

**Review by blg666 — APPROVED** [review]

**Pull request merged**